packages/flutter/lib/src/material/button_theme.dart

```diff
@@ -284,6 +314,197 @@ class ButtonThemeData extends Diagnosticable {
   /// This property only affects [DropdownButton] and its menu.
   final bool alignedDropdown;
 
+  final Color buttonColor;
+  final Color disabledColor;
+  final Color highlightColor;
+  final Color splashColor;
+  final ColorScheme colorScheme;
+  final MaterialTapTargetSize materialTapTargetSize;
+
+  Brightness getBrightness(MaterialButton button) {
+    return button.colorBrightness ?? colorScheme.brightness;
+  }
+
+  ButtonTextTheme getTextTheme(MaterialButton button) {
+    return button.textTheme ?? textTheme;
+  }
+
+  bool _isFlatButton(MaterialButton button) {
```

Comment:
In general, I'm a little skeptical of string based APIs. I can definitely see the value in essentially having a tag/selector like value for determining theme values though. Is there an advantage over using an enumeration - besides visibility?

Reply:
I agree. Enums have some serious disadvantages which put me off finding a better solution (than Strings):

- They can't be extended. The "type" of subclasses of MaterialButton will be limited to the enum values we define.
- They're difficult for us to extend. Adding an enum value later will break code that switches on the value and doesn't include a default clause.